src/model/model.py

```diff
@@ -90,6 +98,32 @@ def save(self, data, path):
             imageio.imwrite(path, data)
         return "saved as {}".format(path)
 
+    def extract_channel(self, im, red=True, green=True, blue=True, alpha=True):
+        """
+        extract channel from image
+
+        Parameters
+        ----------
+        im: 2D numpy array
+        red, green, blue, alpha: bool, default=True
+            if False, drop this channel
+
+        Return
+        ------
+        result: 2D numpy array
+
+        """
+        if im.shape[2] == 4:
+            selected_channels = [i for i, ch in enumerate([red, green, blue, alpha]) if ch]
+        elif im.shape[2] == 3:
+            selected_channels = [i for i, ch in enumerate([red, green, blue]) if ch]
+        if len(selected_channels) == 1:
+            selected_channels = selected_channels[0]
+        elif len(selected_channels) == 0:
+            raise ValueError("No channel has been selected")
```

Comment:
In this case, you don't handle the wase when several channels are asked, it may cause a bug in line 124 no?
Do this function is intended to extract ONLY ONE channel or could extract several (ex; Blue and Red)?
IMO, limiting the function to extract only one channel can be sufficient, what do you think?

Reply:
The function works if one or several channels are asked. So you can select only red and green if you want. 
I agree we should limit the function to extract only one channel...

Reply:
And now its done